x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPutTransformAction.java

```diff
@@ -206,12 +207,24 @@ protected void masterOperation(Task task, Request request, ClusterState clusterS
             return;
         }
 
+        List<SourceDestValidator.SourceDestValidation> validations;
```

Comment:
nit: is it possible to avoid the almost duplicate code?

(don't do, if you see a way to re-factor this for the upcoming validation refactorings)

Reply:
Done. I did a small refactoring:
1. I introduced `SourceDestValidations.getValidations` and `SourceDestValidations.getValidationsForPreview` methods. This also allowed making the static fields in the `SourceDestValidations` class `private`.
2. I made `TransformConfig` class responsible for creating appropriate additional validation(s).